riscv-executor/src/pil.rs

```diff
@@ -0,0 +1,144 @@
+use std::ops::ControlFlow;
+
+use powdr_ast::{
+    analyzed::{AlgebraicExpression, Analyzed, Identity, PermutationIdentity},
+    parsed::visitor::ExpressionVisitable,
+};
+use powdr_number::FieldElement;
+
+pub fn links_from_pil<F: FieldElement>(pil: &Analyzed<F>) -> Vec<Identity<F>> {
+    pil.identities
+        .iter()
+        .filter(|id| matches!(id, Identity::Permutation(_) | Identity::Lookup(_)))
```

Comment:
I guess later we'll also need to support the Phantom versions, and/or bus sends/receives. It might actually be simpler at that point.

Reply:
yeah, I didn't really look into it here, but should not be to bad I imagine